kevlar/include/kevlar_bits/model/binomial/berry_inla.hpp

```diff
@@ -0,0 +1,372 @@
+#pragma once
+#include <Eigen/Dense>
+#include <algorithm>
+#include <kevlar_bits/model/base.hpp>
+#include <kevlar_bits/model/binomial/common/fixed_n_default.hpp>
+#include <kevlar_bits/model/fixed_single_arm_size.hpp>
+#include <kevlar_bits/util/algorithm.hpp>
+#include <kevlar_bits/util/legendre.hpp>
+#include <kevlar_bits/util/macros.hpp>
+#include <kevlar_bits/util/math.hpp>
+#include <kevlar_bits/util/types.hpp>
+
+namespace kevlar {
+namespace model {
+namespace binomial {
+
+template <class ValueType, int ARMS>
+struct BerryINLA : FixedSingleArmSize, ModelBase<ValueType> {
+    using arm_base_t = FixedSingleArmSize;
+    using base_t = ModelBase<ValueType>;
+    using typename base_t::value_t;
+
+   private:
+    using vec_t = colvec_type<value_t>;
+    using mat_t = mat_type<value_t>;
+
+    static constexpr double mu_0 = -1.34;
+
+    const size_t n_arm_samples;
+    const vec_t efficacy_thresholds;
+
+    // variables dependent solely on the quadrature points (independent of the
+    // data).
+    const vec_t quad_wts;
+    const mat_t cov;
+    const mat_t neg_precQ;
+    const vec_t logprecQdet;
+    const vec_t logprior;
+    const value_t opt_tol2;
+    const value_t logit_p1;
+
+   public:
+    template <class _GenType, class _ValueType, class _UIntType,
+              class _GridRangeType>
+    struct SimGlobalState;
+
+    template <class _GenType, class _ValueType, class _UIntType,
+              class _GridRangeType>
+    using sim_global_state_t =
+        SimGlobalState<_GenType, _ValueType, _UIntType, _GridRangeType>;
+
+    template <class _GridRangeType>
+    using kevlar_bound_state_t = KevlarBoundStateFixedNDefault<_GridRangeType>;
+
+    BerryINLA(size_t n_arm_samples, const Eigen::Ref<const vec_t>& cv,
+              const Eigen::Ref<const vec_t>& efficacy_thresholds,
+              const Eigen::Ref<const vec_t>& quad_wts,
```

Comment:
I left a bunch of these parameters generated in Python and passed in as arguments to the constructor.